### PR TITLE
Add filter bar to collection view

### PR DESCRIPTION
**Pull request by SektorDV**
I want to merge this change because it replaces the search bar in collection list views with a filter bar.

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3695 Collection list adjustments](https://app.clickup.com/t/2549495/SALEOR-3695)

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/components/AvailabilityStatusLabel/AvailabilityStatusLabel.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { useIntl } from "react-intl";
 
 import { messages } from "./messages";
 
-export const ProductAvailabilityStatusLabel = ({ channel }) => {
+export const AvailabilityStatusLabel = ({ channel, type }) => {
```
Comment (jwm0) [inline src/components/AvailabilityStatusLabel/AvailabilityStatusLabel.tsx:8]:
```suggestion
export const AvailabilityStatusLabel = ({ channel, type = "" }) => {
```

File: src/components/AvailabilityStatusLabel/AvailabilityStatusLabel.tsx [hunk 2]
```diff
@@ -18,7 +18,8 @@ export const ProductAvailabilityStatusLabel = ({ channel }) => {
             : messages.willBePublished
           : messages.unpublished,
         {
-          date: localizeDate(channel.publicationDate, "L")
+          date: localizeDate(channel.publicationDate, "L"),
+          type: type || ""
```
Comment (jwm0) [inline src/components/AvailabilityStatusLabel/AvailabilityStatusLabel.tsx:22]:
```suggestion
          type
```

File: src/products/components/ProductList/ProductList.tsx [hunk 3]
```diff
@@ -376,7 +375,10 @@ export const ProductList: React.FC<ProductListProps> = props => {
                       {(!product && <Skeleton />) ||
                         (!product?.channelListings?.length && "-") ||
                         (product?.channelListings !== undefined && channel ? (
-                          <ProductAvailabilityStatusLabel channel={channel} />
+                          <AvailabilityStatusLabel
+                            channel={channel}
+                            type="product"
```
Comment (jwm0) [inline src/products/components/ProductList/ProductList.tsx:380]: `type` has to be localized as well. 
```suggestion
                            type={intl.formatMessage({ defaultMessage: "product" })}
```

File: src/collections/components/CollectionList/CollectionList.tsx [hunk 4]
```diff
@@ -178,19 +186,20 @@ const CollectionList: React.FC<CollectionListProps> = props => {
...
-                  {collection && !collection?.channelListings?.length ? (
-                    "-"
-                  ) : collection?.channelListings !== undefined ? (
-                    channel ? (
+                  {(!collection && <Skeleton />) ||
+                    (!collection?.channelListings?.length && "-") ||
+                    (collection?.channelListings !== undefined && channel ? (
+                      <AvailabilityStatusLabel
+                        channel={channel}
+                        type="collection"
```
Comment (jwm0) [inline src/collections/components/CollectionList/CollectionList.tsx:194]:
```suggestion
                        type={intl.formatMessage({ defaultMessage: "collection" })}
```

**Review by jwm0 — CHANGES_REQUESTED** [review]

**Review by jwm0 — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

File: src/components/AvailabilityStatusLabel/messages.ts [hunk 5]
```diff
@@ -3,14 +3,14 @@ import { defineMessages } from "react-intl";
 export const messages = defineMessages({
   published: {
     defaultMessage: "Published on {date}",
-    description: "product publication date"
+    description: "{type} publication date"
```
Comment (dominik-zeglen) [inline src/components/AvailabilityStatusLabel/messages.ts:6]: It's not templated so translator will see this string exactly as you've written it

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

File: src/components/AvailabilityStatusLabel/messages.ts (hunk 5 above)
Comment (patrys) [reply src/components/AvailabilityStatusLabel/messages.ts:6]: We can't do it like this. If we reuse the same message for different objects, there's no way for translators to differentiate between gendered nouns ("produkt opublikowany" vs. "wersja opublikowana").

Also, make sure `{date}` is actually a date and not date&time ("opublikowany dnia 12:31 w czwartek").

**Review by dominik-zeglen — APPROVED** [review]

**Pull request merged**